front-end/src/components/productView/ProductView.js

```diff
@@ -71,23 +79,37 @@ const ProductView = () => {
         productService.getSingelProduct(id, setTitle, setCategory, setSize, setColor, setPrice, setImageUrl)
     }, [id])
 
+    const addToBasket = () => {
+        const productToBasket = {
+            _id: id, title, category, size, color, price, image: imageUrl
+        }
+        if (basketProducts.some(product => product._id === id)) {
+            alert('Allready in basket')
+        }
+        else {
+            productsIds.push(id);
+            userServices.addToBasket(dispatch, user?._id, productsIds, productToBasket);
+        }
+    }
+
     const deleteProduct = () => {
         productService.deleteProduct(id, title, history);
     }
 
     return (
         <Container>
-            <div className="letf__container">
+            <div className="letfContainer">
                 <p>{category}</p>
                 <img src={imageUrl} alt="product-img" />
             </div>
-            <div className="right__container">
+            <div className="rightContainer">
                 <h2>{title}</h2>
                 <p>size: {size}</p>
                 <p>color: {color}</p>
                 <p className="price">price: {price} $</p>
                 <div>
-                    <button onClick={() => {history.push(`/edit/${id}`)}}>Edit</button>
```

Comment:
Create a separate function and passed it on onClick, in case to reuse it on line 112